datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/useGetEntitySchema.tsx

```diff
@@ -0,0 +1,24 @@
+import { cloneDeep } from 'lodash';
+import { EntityType } from '../../../../../../types.generated';
+import { useEntityData } from '../../../EntityContext';
+import { useGetDatasetSchemaQuery } from '../../../../../../graphql/dataset.generated';
+import { combineEntityDataWithSiblings, useIsSeparateSiblingsMode } from '../../../siblingUtils';
+
+export const useGetEntityWithSchema = () => {
+    const { urn, entityData, entityType } = useEntityData();
+    // Load the dataset schema lazily.
+    const { data: rawData, loading } = useGetDatasetSchemaQuery({
+        variables: {
+            urn,
+        },
+        skip: entityType !== EntityType.Dataset,
+        fetchPolicy: 'cache-first',
+    });
+    const isHideSiblingMode = useIsSeparateSiblingsMode();
+    // Merge with sibling information as required.
+    const combinedData = rawData && !isHideSiblingMode ? combineEntityDataWithSiblings(cloneDeep(rawData)) : rawData;
```

Comment:
does the `cloneDeep` here just ensure we don't mess with the original `rawData` object in this method? is that necessary in the other places we call `combineEntityDataWithSiblings` or only here because it's `cache-first`?

Reply:
it's because of cache! apollo hates it when we f with the objects it has in cache!